sample-tracking-status-overview-app/src/main/groovy/life/qbic/portal/sampletracking/resource/ResourceMessage.groovy

```diff
@@ -0,0 +1,28 @@
+package life.qbic.portal.sampletracking.resource
+
+import life.qbic.portal.sampletracking.communication.Message
+
+
+/**
+ * <p>A message that communicates changes in a resource of type T</p>
+ *
+ * @param <T>  the data type stored in the resource
+ * @since 1.0.0
+ */
+abstract class ResourceMessage<T> extends Message {
+    public Optional<T> payload
+    private final MessageType type
+
+    ResourceMessage(T payload, MessageType type) {
```

Comment:
We could also choose `V` here. This is the payload data type (for example `Project`)